src/modules/home/components/property/property-card.tsx

```diff
@@ -14,16 +16,37 @@ import './swiper.css';
 
 import { EffectFade, Navigation, Pagination } from 'swiper/modules';
 
-export function ApartmentCard(apartment: Accommodation) {
+export function ApartmentCard({ id, name, location, pricePerNight, images }: Accommodation) {
+  const [isLiked, setIsLiked] = useState(false);
+  const navidate = useNavigate();
+  useEffect(() => {
+    const likedStatus = localStorage.getItem(`liked-${id}`);
+    setIsLiked(likedStatus === 'true');
+  }, [id]);
+
   const handleFavoriteClick = (event: React.MouseEvent) => {
     event.stopPropagation();
     event.preventDefault();
+    const isUserExist = localStorage.getItem(`isUserLoggedIn`);
+
+    if (!isUserExist) {
+      navidate('/login');
+      return;
+    }
+
+    const newLikedStatus = !isLiked;
```

Comment:
I don't think new variable is needed here

Reply:
changed